react/week4/my-app/src/components/FetchUser.js

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useContext } from "react";
+import { stateContext } from "../App.js";
+
+function FetchUser() {
+  const {
+    query,
+    user,
+    setUser,
+    error,
+    setError,
+    loading,
+    setLoading
+  } = useContext(stateContext);
+
+  useEffect(() => {
+    const URL = `https://api.github.com/search/users?q=${query}`;
+    (async () => {
+      try {
+        const response = await fetch(`${URL}`);
+        const json = await response.json();
+        const result = json;
+        setUser(result);
+        setLoading(false);
+        setError(result.message);
+      } catch (error) {
+        console.log(error);
+      }
+    })();
+  }, [query, setUser, error, setError, setLoading]);
```

Comment:
Right now, whenever any of these constants change, you will run `fetch`. As I understand, you need to run it again only when `query` changes.